### ISSUE TITLE
Command Runner support for dnx beta7

### ISSUE THREAD

**Issue by david-driscoll** [OmniSharp/omnisharp-atom#488]
For beta7 `dnx . <command>` is not `dnx <command>`, we need to adjust the command runner to use the legacy command for beta 4,5,6, but then use the new command going forward.

http://blogs.msdn.com/b/webdev/archive/2015/09/02/announcing-availability-of-asp-net-5-beta7.aspx

**Comment by Mpdreamz** [comment]
are we going to indefinitely support < beta7 ? I'm +1 on just moving over to the beta7 ways of doing things and moving folks over.

**Comment by david-driscoll** [comment]
I wouldn't say indefinitely, but if we do it right, we can remove the old
code around beta8 or rc1

**Comment by melcloud** [comment]
Had this change been included in v0.24.6? Because if I use beta7, all test command short cut (e.g. CTRL+R, CTRL+A) would throw exception:

```
System.InvalidOperationException: Unable to load application or execute command '.'. Available commands: test.
   at Microsoft.Dnx.ApplicationHost.Program.ThrowEntryPointNotfoundException(DefaultHost host, String applicationName, Exception innerException)
   at Microsoft.Dnx.ApplicationHost.Program.ExecuteMain(DefaultHost host, String applicationName, String[] args)
   at Microsoft.Dnx.ApplicationHost.Program.Main(String[] args)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Dnx.Runtime.Common.EntryPointExecutor.Execute(Assembly assembly, String[] args, IServiceProvider serviceProvider)
   at Microsoft.Dnx.Host.Bootstrapper.RunAsync(List`1 args, IRuntimeEnvironment env, FrameworkName targetFramework)
   at Microsoft.Dnx.Host.RuntimeBootstrapper.ExecuteAsync(String[] args, FrameworkName targetFramework)
   at Microsoft.Dnx.Host.RuntimeBootstrapper.Execute(String[] args, FrameworkName targetFramework)
01201 ProjectsSampleTest.cs13:12
```

Thanks

**Comment by david-driscoll** [comment]
Ill look into it.

On Wed, Sep 9, 2015 at 9:25 AM Yin Zhang <notifications@github.com> wrote:

> Had this change been included in v0.24.6? Because if I use beta7, all test
> command short cut (e.g. CTRL+R, CTRL+A) would throw exception:
>
> System.InvalidOperationException: Unable to load application or execute command '.'. Available commands: test.
>    at Microsoft.Dnx.ApplicationHost.Program.ThrowEntryPointNotfoundException(DefaultHost host, String applicationName, Exception innerException)
>    at Microsoft.Dnx.ApplicationHost.Program.ExecuteMain(DefaultHost host, String applicationName, String[] args)
>    at Microsoft.Dnx.ApplicationHost.Program.Main(String[] args)
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at Microsoft.Dnx.Runtime.Common.EntryPointExecutor.Execute(Assembly assembly, String[] args, IServiceProvider serviceProvider)
>    at Microsoft.Dnx.Host.Bootstrapper.RunAsync(List`1 args, IRuntimeEnvironment env, FrameworkName targetFramework)
>    at Microsoft.Dnx.Host.RuntimeBootstrapper.ExecuteAsync(String[] args, FrameworkName targetFramework)
>    at Microsoft.Dnx.Host.RuntimeBootstrapper.Execute(String[] args, FrameworkName targetFramework)
> 01201 ProjectsSampleTest.cs13:12
>
> Thanks
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OmniSharp/omnisharp-atom/issues/488#issuecomment-138908207>
> .
>
-- 
Thanks,
David
